File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/club/enumeration/ClubStatus.java
package b1nd.dodam.domain.rds.club.enumeration;

import java.util.ArrayList;
import java.util.List;

public enum ClubStatus {
    ALLOWED,
    PENDING,
    REJECTED,
    WAITING,
    DELETED;

    public static List<ClubStatus> getNotAllowedStatuses() {
        List<ClubStatus> statuses = new ArrayList<>();
        statuses.add(WAITING);
        statuses.add(REJECTED);
        statuses.add(DELETED);
        return statuses;
    }
}

File: services/service-club/src/main/kotlin/com/b1nd/dodamdodam/club/domain/club/enumeration/ClubStatus.kt
package com.b1nd.dodamdodam.club.domain.club.enumeration

enum class ClubStatus {
    ALLOWED,
    PENDING,
    REJECTED,
    WAITING,
    DELETED,
    ;

    companion object {
        fun getNotAllowedStatuses(): List<ClubStatus> = listOf(WAITING, REJECTED, DELETED)
    }
}
